qiskit/algorithms/optimizers/cmaes.py

```diff
@@ -0,0 +1,400 @@
+
+    A new covariance matrix is also computed taking into account not only the current population,
+    but also the previous covariance matrix.
+
+    The optimization is stopped either at the begining of the optimization process if the initial
+    parameters were not well chosen, or once the objective function stops decreasing for a given
+    number of generations. There are several criteria that can be manually set by the user, and
+    are well described in the paper.
+
+    One evaluation with a funciton evaluation outlier with a significantly low value is not
+    enough to stop the optimization process or significantly disturb the position of the centroid,
+    and an outlier with a significantly high value would simply be ignored by the algorithm.
+    Therefore, this optimizer is resistant to noise in the objective function and can be used with
+    non-smooth functions.
+    """
+
+    def __init__(
+        self,
+        maxiter: int = 100,
+        callback: Optional[CALLBACK] = None,
+        sigma_0: float = 0.5,
+        weights: Optional[np.ndarray] = None,
+        termination_criteria: Optional[Dict[str, Optional[float]]] = None,
+    ) -> None:
+        """
+        Args:
+            maxiter: Maximum number of iterations.
+            callback: A callback function to be called after each iteration when using :meth:`~.minimize`.
+            sigma_0: Standard deviation for the sampling of the first generation.
+            weights: Weights used to compute the next centroid and covariance matrix. If ``None``, they
+            will take a default value. All components must be positive, arranged in decreasing order,
+            and add up to one.
+            termination_criteria: Dictionary with the terimnation criteria to be used and the tolerances
+            for each of them. The different options and their default values are:
+
+            * NoEffectAxis: 0
+            * NoeffectCoord: 1e4
+            * TolFun: 1e-12
+            * TolX: :math:`\sigma`
+            * NoEffectCoord: 0
+            * ConditionCov: 1e14
+            * EqualFunValues: 0
+            * Stagnation : WIP
+        """
+
+        super().__init__(maxiter=maxiter, callback=callback)
+        self._state: CMAESState = None
+        self.N = None
+        self.lmbda = None
+        self.weights = weights
+        self.mu = None
+        self.mueff = None
+        self.sigma_0 = sigma_0
+        self.cc = None  # Learing rate for evolution path pc.
+        self.cs = None
+        self.c1 = None  # Learning rate for rank 1 updates. This updates make the next sampling more likely in the direction of the best sampled point from last generateion.
+        self.cmu = None  # Covariance Matrix Learning rate
+        self.damps = None
+        self.chiN = None
+        self.termination_criteria = termination_criteria
+
+    def ask(self) -> CMAES_AskObject:
+        """ """
```

Comment:
Is there are reason for an empty docstring here - this should inherit the docstring of the SteppableOptimizer where its defined and which this is an implementation of